File: backend/src/main/java/com/example/entity/CategoryMaster.java
package com.example.entity;

import jakarta.persistence.*;
import lombok.*;

/**
 * Maps the `category_master` table (from your teacher's original design).
 *
 * Owned by Module 5 (Catalog); defined here because ProductMaster has a
 * foreign key to it and Module 6 loads products.
 */
@Entity
@Table(name = "category_master")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CategoryMaster {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "catmaster_id")
    private Integer catmasterId;

    @Column(name = "cat_id", nullable = false, length = 3)
    private String catId;

    /** '^' marks a root category in the seed data; otherwise it is a parent's cat_id. */
    @Column(name = "subcat_id", length = 3)
    private String subcatId;

    @Column(name = "cat_name", nullable = false, length = 255)
    private String catName;

    @Column(name = "cat_image_path", length = 255)
    private String catImagePath;

    @Column(name = "flag", nullable = false)
    private Boolean flag;
}

File: dotnet-backend/Emart.Api/Models/CategoryMaster.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Emart.Api.Models
{
    [Table("category_master")]
    public class CategoryMaster
    {
        [Key]
        [Column("catmaster_id")]
        public int CatmasterId { get; set; }

        [Required]
        [Column("cat_id")]
        [MaxLength(3)]
        public string CatId { get; set; } = string.Empty;

        [Column("subcat_id")]
        [MaxLength(3)]
        public string? SubcatId { get; set; }

        [Required]
        [Column("cat_name")]
        [MaxLength(255)]
        public string CatName { get; set; } = string.Empty;

        [Column("cat_image_path")]
        [MaxLength(255)]
        public string? CatImagePath { get; set; }

        [Column("flag")]
        public bool Flag { get; set; }
    }
}
